client/packages/programs/src/JsonForms/components/AdherenceStatus.tsx

```diff
@@ -70,25 +70,15 @@ const UIComponent = (props: ControlProps) => {
 
   // fetch current encounter
   const encounterId = useEncounter.utils.idFromUrl();
-  const { data: currentEncounter, mutate: fetchEncounter } =
-    useEncounter.document.byId(encounterId);
-  useEffect(() => {
-    fetchEncounter();
-  }, [encounterId]);
+  const { data: currentEncounter } = useEncounter.document.byId(encounterId);
 
   // fetch previous encounter
-  const { data: previousEncounter, mutate: fetchPreviousEncounter } =
-    useEncounter.document.previous(
-      currentEncounter?.patient.id ?? '',
-      currentEncounter?.startDatetime
-        ? new Date(currentEncounter?.startDatetime)
-        : new Date()
-    );
-  useEffect(() => {
-    if (currentEncounter) {
-      fetchPreviousEncounter();
-    }
-  }, [currentEncounter]);
+  const { data: previousEncounter } = useEncounter.document.previous(
+    currentEncounter?.patient.id,
+    currentEncounter?.startDatetime
+      ? new Date(currentEncounter?.startDatetime)
+      : new Date()
+  );
```

Comment:
Github won't let me comment in the right place (cos it's unchanged), but you have another `useEffect` that's doing nothing at line 122 😆

Reply:
Ups what were I doing??